documentation/json_output_reference.md

```diff
@@ -0,0 +1,763 @@
+
+  - `action` (string):
+    The type of action that was executed. This value is always `download`.
+
+  - `object` (string):
+    The path to the file on the target.
+
+  - `status` (string):
+    Whether the download completed successfully. This value is `success` if the
+    download is successful and `failure` otherwise.
+
+  - `value` (object):
+    The value returned by the download.
+
+    - `_output` (string):
+      A human-readable message describing where the file was downloaded to on the
+      controller.
+
+    - `path` (string):
+      The path to the downloaded file on the controller.
+
+- `target_count` (number):
+  The number of targets the file was downloaded from.
+
+For example:
+
+```json
+{
+  "items": [
+    {
+      "target": "nix",
+      "action": "download",
+      "object": "/etc/ssh/sshd_config",
+      "status": "success",
+      "value": {
+        "path": "/Users/bolt/.puppetlabs/bolt/downloads/nix/sshd_config",
+        "_output": "Downloaded 'nix:/etc/ssh/sshd_config' to '/Users/bolt/.puppetlabs/bolt/downloads/nix'"
+      }
+    }
+  ],
+  "target_count": 1,
+  "elapsed_time": 1
+}
+```
+
+## `file upload`
+
+This command outputs an object. It uses the following keys:
+
+- `elapsed_time` (number):
+  The total time, in seconds, that it took to upload the file to all targets.
+
+- `items` (array of objects):
+  The results of uploading the file to each target.
+
+  - `action` (string):
+    The type of action that was executed. This value is always `upload`.
+
+  - `object` (string):
+    The path to the file on the controller.
+
+  - `status` (string):
+    Whether the upload completed successfully. This value is `success` if the
+    upload is successful and `failure` otherwise.
+
+  - `value` (object):
+    The value returned by the upload.
+
+    - `_output` (string):
+      A human-readable message describing where the file was uploaded to on the
+      target.
+
+- `target_count` (number):
+  The number of targets the file was uploaded to.
+
+For example:
+
+```json
+{
+  "items": [
+    {
+      "target": "nix",
+      "action": "upload",
+      "object": "files/configure.sh",
+      "status": "success",
+      "value": {
+        "_output": "Uploaded 'files/configure.sh' to 'nix:./configure.sh'"
+      }
+    }
+  ],
+  "target_count": 1,
+  "elapsed_time": 1
+}
+
+```
+
+## `group show`
+
+This command outputs an object. It uses the following keys:
+
+- `count` (number):
+  The number of groups in the inventory.
+
+- `groups` (array of strings):
+  The names of all groups in the inventory.
+
+For example:
+
+```json
+{
+  "count": 3,
+  "groups": [
+    "all",
+    "linux",
+    "windows"
+  ]
+}
+```
+
+## `guide`
+
+This command outputs an object. It uses the following key:
+
+- `topics` (array of strings):
+  The list of available topics.
+
+For example:
+
+```json
+{
+  "topics": [
+    "inventory",
+    "logging",
+    "module"
+  ]
+}
+```
+
+## `inventory show`
```

Comment:
Hm, I think this is one of the trickier commands given the note about different forms of the command having different outputs, since `--detail` only has `targets` and `count`. Maybe including `adhoc` and `inventory` in a separate subheader that's explicitly without `--detail` is the simplest? I think the risk is that I suspect `inventory show --detail` is pretty useful, especially using JSON, and it'd be frustrating to see the keys documented but then have them not be in the output and not know why they weren't in the output.